### PR TITLE
Angular 9 / 10

### PR DESCRIPTION
**Pull request by mokipedia**
Update demo app to Angular10
Small fixes for compatibility:

- module with providers must be generic since ng10
- angular 9+ requires node 10.13
- peer dependencies more version agnostic (Docs still mention ng4, so the lib now allows all ng versions since 4)

### CODE REVIEW

File: projects/lib/package.json [hunk 1]
```diff
@@ -14,21 +14,24 @@
...
   },
   "dependencies": {
     "tslib": "^2.0.0"
-},
+  },
   "peerDependencies": {
-    "@angular/common": "^9.1.0",
-    "@angular/core": "^9.1.0"
+    "@angular/common": "^4.0.0 || ^5.0.0 || ^6.0.0 || ^7.0.0 || ^8.0.0 || ^9.0.0 || ^10.0.0",
+    "@angular/core": "^4.0.0 || ^5.0.0 || ^6.0.0 || ^7.0.0 || ^8.0.0 || ^9.0.0 || ^10.0.0"
```
Comment (jaibatrik) [inline projects/lib/package.json:35]: I am little bit apprehensive about this. I understand that the documentation says that is supports Angular 4+, but my idea was to have a compatibility table with older Angular versions pointing to older releases of this library. The newest version of the library would always support the newest version of Angular.

I am not opposed to this though. I'll be rather happy to merge this if the library still works with all these older versions of Angular.

Comment (mokipedia) [reply projects/lib/package.json:35]: I admit, I am also not fond of this, though it reflects the current setup.
Your suggestion to have it narrowed to one or 2 version of angular per library release is definately preferable and should be the way forward.
I will try and test the latest version with a new angular 4 setup later, to find out what the latest version is that still works with 3.1.1

Comment (mokipedia) [reply projects/lib/package.json]: so i tried it out, v2.x works at least down to 5.
3.x seems to need 9 or 10 so i would adjust this to from now on require 9+

Comment (jaibatrik) [reply projects/lib/package.json]: Thanks, that helps a lot 👍

**Comment by jaibatrik** [comment]
I'll update the README and publish to npm. Thanks for the contribution 🥇

**Comment by kapilSoni101** [comment]
@jaibatrik : sir  i am implementing social login but i got below error in the console can you please help me?

**Error: Uncaught (in promise): Login providers not ready yet. Are there errors on your console?**

File: projects/lib/package.json (hunk 1 above)
Comment (kapilSoni101) [reply projects/lib/package.json]: sir i am using Angular 9 and using 3+ version but initially **i got error: "idpiframe_initialization_failed"** error in the console can you please help me?

**Comment by rushiAquil** [comment]
@kapilSoni101  @jaibatrik  sir same error giving to me if issue solved please help me
Error: Uncaught (in promise): Login providers not ready yet. Are there errors on your console?

File: projects/lib/package.json (hunk 1 above)
Comment (deelanM) [reply projects/lib/package.json]: Hi, does anyone know what version works with Angular 4 ? 
2.0.0 doesn't do it

Comment (jaibatrik) [reply projects/lib/package.json]: I think 1.X versions were made for Angular 4.

**Comment by abdulqayyum935** [comment]
Hi All, 
I'm seeing the below message.

Could not resolve dependency:
npm ERR! peer @angular/common@"^9.0.0 || ^10.0.0 || ^11.0.0" from angularx-social-login@3.5.7
npm ERR! node_modules/angularx-social-login
npm ERR!   angularx-social-login@"*" from the root project

Here are the other details of my development environment.

Angular CLI: 12.0.1
Node: 16.2.0
Package Manager: npm 7.13.0
OS: win32 ia32

Angular: 12.0.1
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1200.1
@angular-devkit/build-angular   12.0.1
@angular-devkit/core            12.0.1
@angular-devkit/schematics      12.0.1
@schematics/angular             12.0.1
rxjs                            6.6.7
typescript                      4.2.4

**Comment by mokipedia** [comment]
> Hi All, 
> I'm seeing the below message.
> 
> Could not resolve dependency:
> npm ERR! peer @angular/common@"^9.0.0 || ^10.0.0 || ^11.0.0" from angularx-social-login@3.5.7
> npm ERR! node_modules/angularx-social-login
> npm ERR!   angularx-social-login@"*" from the root project
> 
> Here are the other details of my development environment.
> 
> Angular CLI: 12.0.1
> Node: 16.2.0
> Package Manager: npm 7.13.0
> OS: win32 ia32
> 
> Angular: 12.0.1
> ... animations, cli, common, compiler, compiler-cli, core, forms
> ... platform-browser, platform-browser-dynamic, router
> 
> Package                         Version
> ---------------------------------------------------------
> @angular-devkit/architect       0.1200.1
> @angular-devkit/build-angular   12.0.1
> @angular-devkit/core            12.0.1
> @angular-devkit/schematics      12.0.1
> @schematics/angular             12.0.1
> rxjs                            6.6.7
> typescript                      4.2.4
> 

Have you tried `npm install --legacy-peer-deps`?

**Comment by abdulqayyum935** [comment]
@mokipedia  Yeah I have run the command inside the project folder and it returned "Unsupported engine".

**Comment by mokipedia** [comment]
Do you have the same problem on a fresh project with angular 12? Looks to mee as if angular kind of warns you about not actively supporting node 16

**Comment by abdulqayyum935** [comment]
@mokipedia  Thanks for the guidance. 
The package is now installed with the following command.
$ npm i angularx-social-login --legacy-peer-deps

**Comment by eddzmaciel** [comment]
Hi guys, I am getting this error after installation and  running in an Angular 11 Project:
```
   Error: node_modules/angularx-social-login/socialauth.service.d.ts:59:21 - error TS2694: Namespace '"/Users/edd/docs/dev/reset/haztrato-project/UiHazTrato/node_modules/@angular/core/core"' has no exported member 'ɵɵFactoryDeclaration'.
    
    59     static ɵfac: i0.ɵɵFactoryDeclaration<SocialAuthService, never>;
                           ~~~~~~~~~~~~~~~~~~~~
    node_modules/angularx-social-login/socialauth.service.d.ts:60:22 - error TS2694: Namespace '"/Users/edd/docs/dev/reset/haztrato-project/UiHazTrato/node_modules/@angular/core/core"' has no exported member 'ɵɵInjectableDeclaration'.
    
    60     static ɵprov: i0.ɵɵInjectableDeclaration<SocialAuthService>;
                            ~~~~~~~~~~~~~~~~~~~~~~~
    node_modules/angularx-social-login/sociallogin.module.d.ts:11:21 - error TS2694: Namespace '"/Users/edd/docs/dev/reset/haztrato-project/UiHazTrato/node_modules/@angular/core/core"' has no exported member 'ɵɵFactoryDeclaration'.
    
    11     static ɵfac: i0.ɵɵFactoryDeclaration<SocialLoginModule, [{ optional: true; skipSelf: true; }]>;
                           ~~~~~~~~~~~~~~~~~~~~
    node_modules/angularx-social-login/sociallogin.module.d.ts:12:21 - error TS2694: Namespace '"/Users/eddzmaciel/Documents/dev/RESET - SYSTEMS/haztrato-project/UiHazTrato/node_modules/@angular/core/core"' has no exported member 'ɵɵNgModuleDeclaration'.
    
    12     static ɵmod: i0.ɵɵNgModuleDeclaration<SocialLoginModule, never, [typeof i1.CommonModule], never>;
                           ~~~~~~~~~~~~~~~~~~~~~
    node_modules/angularx-social-login/sociallogin.module.d.ts:13:21 - error TS2694: Namespace '"/Users/edd/docs/dev/reset/haztrato-project/UiHazTrato/node_modules/@angular/core/core"' has no exported member 'ɵɵInjectorDeclaration'.
    
    13     static ɵinj: i0.ɵɵInjectorDeclaration<SocialLoginModule>;

```

Any ideas for this ????

**Comment by adnanrafique** [comment]
@eddzmaciel were you able to resolve it? I am getting the same error with angular version 11 and angularx-social-login version 4.0.1.

**Comment by Bufarhan** [comment]
me too! the same error with angular 13

**Comment by anton-nikitsiuk** [comment]
> @eddzmaciel were you able to resolve it? I am getting the same error with angular version 11 and angularx-social-login version 4.0.1.

Looks like for Angular 11 we should use a previous version - 3.5.7
image

It works for me

**Comment by mokipedia** [comment]
please open an issue, this is an already merged and frankly quite old pull request

**Pull request merged**